catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,313 @@
+
+    public void setKeepDeletedMetacards(Boolean keepDeletedMetacards) {
+        this.keepDeletedMetacards = keepDeletedMetacards;
+    }
+
+    public Boolean getKeepDeletedMetacards() {
+        return keepDeletedMetacards;
+    }
+
+    public void setMetacardTransformerId(String metacardTransformerId) {
+        this.metacardTransformerId = metacardTransformerId;
+        this.metacardTransformer = lookupTransformerReference();
+    }
+
+    public void setMetacardTransformer(MetacardTransformer metacardTransformer) {
+        this.metacardTransformer = metacardTransformer;
+    }
+
+    public String getMetacardTransformerId() {
+        return metacardTransformerId;
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        LOGGER.debug("Properties : {}", properties);
+
+        Object outputDirectory = properties.get(OUTPUT_DIRECTORY_PROPERTY);
+        if (outputDirectory instanceof String && StringUtils.isNotBlank((String) outputDirectory)) {
+            this.outputDirectory = (String) outputDirectory;
+            LOGGER.debug("Updating {} with {}", OUTPUT_DIRECTORY_PROPERTY, outputDirectory);
+        }
+
+        Object metacardTransformerProperty = properties.get(METACARD_TRANSFORMER_ID_PROPERTY);
+        if (metacardTransformerProperty instanceof String && StringUtils.isNotBlank((String) metacardTransformerProperty)) {
+            setMetacardTransformerId((String) metacardTransformerProperty);
+            LOGGER.debug("Updating {} with {}",
+                    METACARD_TRANSFORMER_ID_PROPERTY, metacardTransformerProperty);
+        }
+
+        Object keepDeletedMetacards = properties.get(KEEP_DELETED_METACARDS_PROPERTY);
+        if (keepDeletedMetacards instanceof Boolean) {
+            this.keepDeletedMetacards = (Boolean) keepDeletedMetacards;
+            LOGGER.debug("Updating {} with {}", KEEP_DELETED_METACARDS_PROPERTY,
+                    keepDeletedMetacards);
+        }
+
+        Object folderDepth = properties.get(FOLDER_DEPTH_PROPERTY);
+        if (folderDepth instanceof Integer) {
+            setFolderDepth((Integer) folderDepth);
+            LOGGER.debug("Updating {} with {}", FOLDER_DEPTH_PROPERTY, folderDepth);
+        }
+
+    }
+
+    private void processRequest(ProcessRequest<? extends ProcessResourceItem> processRequest)
+            throws PluginExecutionException {
+        LOGGER.trace("Backing up metadata");
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no outputDirectory.");
+        }
+
+        if (metacardTransformer == null) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no Metacard Transformer found.");
+        }
+
+        List<? extends ProcessResourceItem> processResourceItems = processRequest.getProcessItems();
+        for (ProcessResourceItem processResourceItem : processResourceItems) {
```

Comment:
Can this be a 'processResourceItems.forEach(...)' instead? That would eliminate some branch complexity